pkg/client/gen.go

```diff
@@ -66,6 +68,12 @@ func (*SonobuoyClient) GenerateManifest(cfg *GenConfig) ([]byte, error) {
 		return nil, errors.Wrap(err, "couldn't marshall selector")
 	}
 
+	// Allow nil cfg.Config but avoid dereference errors.
+	conf := &config.Config{}
+	if cfg.Config != nil {
+		conf = cfg.Config
+	}
```

Comment:
We didn't ever use the nested values in the config; instead, there are redundant fields on the genConfig. I created a ticket (https://github.com/heptio/sonobuoy/issues/691) to solve that. Otherwise, the behavior is completely unclear.